src/Bot.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import "./interfaces/ICDP.sol";
+import "@cdp/src/interfaces/external/univ3/IUniswapV3Pool.sol";
+import "@cdp/src/interfaces/external/univ3/INonfungiblePositionManager.sol";
+import "./interfaces/IUniswapV3FlashCallback.sol";
+import "./interfaces/IWETH.sol";
+
+contract Bot is IUniswapV3FlashCallback {
+    address[] public pools;
+
+    /// @notice the structure containing all information needed for flashloan + liquidation
+    /// @param vaultId id of the position in CDP contract
+    /// @param debt amount of the loan (in BOB wei) needed to liquidate the position
+    /// @param nfts all positions nfts
+    /// @param swapAddresses the addresses of other contracts, needed to execute swaps
+    /// @param swapData the data for external calls (abi.encodeWithSelector(selector, arg))
+    /// @param positionManager the address of UniV3 position manager, needed to burn all positions
+    /// @param cdp the address of cdp contract
+    /// @param token the bob address
+    struct FlashCallbackData {
+        uint256 vaultId;
+        uint256 debt;
+        uint256[] nfts;
+        address[] swapAddresses;
+        bytes[] swapData;
+        INonfungiblePositionManager positionManager;
+        ICDP cdp;
+        IERC20 token;
+    }
+
+    constructor(address[] memory pools_) {
+        pools = pools_;
```

Comment:
Let's implement methods that allow adding (necessary) and removing (arguably) pools